src/bio.c

```diff
@@ -167,14 +168,21 @@ void bioCreateLazyFreeJob(lazy_free_fn free_fn, int arg_count, ...) {
     bioSubmitJob(BIO_LAZY_FREE, job);
 }
 
-void bioCreateCloseJob(int fd, int need_fsync) {
+void bioCreateCloseJob(int fd) {
     bio_job *job = zmalloc(sizeof(*job));
     job->fd_args.fd = fd;
-    job->fd_args.need_fsync = need_fsync;
 
     bioSubmitJob(BIO_CLOSE_FILE, job);
 }
 
+void bioCreateCloseAofJob(int fd, long long offset) {
+    bio_job *job = zmalloc(sizeof(*job));
+    job->fd_args.fd = fd;
+    job->fd_args.offset = offset;
+
+    bioSubmitJob(BIO_CLOSE_AOF, job);
+}
```

Comment:
My memory is somewhat hazy as well, but AFAIR that race only matters because of the update to `pot_fsynced_reploff`. In other words it doesn't affect the current OSS code.